sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -344,8 +351,11 @@ class StreamExecution(
       }
     }
     if (hasNewData) {
+      // Current batch timestamp in seconds
+      currentBatchTimestamp = triggerClock.getTimeMillis() / 1000L
       reportTimeTaken(OFFSET_WAL_WRITE_LATENCY) {
-        assert(offsetLog.add(currentBatchId, availableOffsets.toOffsetSeq(sources)),
+        assert(offsetLog.add(currentBatchId,
+          availableOffsets.toOffsetSeq(sources, Some(currentBatchTimestamp.toString))),
```

Comment:
It's better to have a new class like `StreamExecutionMetadata` to store all fields that need to be checkpointed, e.g., `currentBatchTimestamp`, `currentEventTimeWatermark`, and others that we may add in future.